### PR TITLE
Adds routes that return mock groups in default JSON format

### PR DESCRIPTION
**Pull request by elvingm**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1,15 +1,67 @@
...
+    e.Use(mw.Logger())
+    e.Use(mw.Recover())
+
+    // Routes
+    e.Get("/groups", getAllGroups)
+    e.Get("/groups/:id/gifs", getGroupGifs)
+    e.Post("/groups", createGroup)
+    e.Post("/groups/:id/gifs", createGifInGroup)
+
+    e.Run(":1323")
```
Comment (BrandonRomano) [inline main.go]: Generally good practice to bring the port into an environment variable.  Makes deploying the app easier, as most providers (like Heroku) dynamically generate this and you have to use their environment variable!

Check out https://golang.org/pkg/os/#Getenv

**Comment by BrandonRomano** [comment]
Echo is a really awesome framework, and you'll find that it is pretty easy to work with.

I'm a big fan of the work you've already put into the README, that's always a great starting point.

My only suggestion (beyond the line comment) is to check out a dependency manager tool for Go (and also add in the README how to get set up).  We like to use [Godep](https://github.com/tools/godep) (which works best with Heroku), and [Gom](https://github.com/mattn/gom) (which feels most like other languages dependency mangers like NPM).  You can pick either one (or even another if you find one that more suits your needs).

**Comment by elvingm** [comment]
Hey @BrandonRomano, great to meet you and thanks a lot for the helpful comments! 

I'm new to Go and Redis so the feedback is helpful to reveal some of the better practices I may be unaware of as I hack away. Will look to incorporate your suggestions with forthcoming PR's and adjust this current one (like your highlight of using `interface{}` rather than `[]Group` -- thanks for pointing that out).

**Comment by elvingm** [comment]
:+1: merging now, since implemented @BrandonRomano's suggestions.

**Pull request merged**